examples/middleware_factory/src/getting_started_middleware_with_params_function.py

```diff
@@ -0,0 +1,59 @@
+import base64
+from dataclasses import dataclass, field
+from typing import Any, Callable, List
+from uuid import uuid4
+
+from aws_lambda_powertools.middleware_factory import lambda_handler_decorator
+from aws_lambda_powertools.utilities.jmespath_utils import envelopes, extract_data_from_envelope
+from aws_lambda_powertools.utilities.typing import LambdaContext
+
+
+@dataclass
+class Booking:
+    days: int
+    date_from: str
+    date_to: str
+    hotel_id: int
+    country: str
+    city: str
+    guest: dict
+    booking_id: str = field(default_factory=lambda: f"{uuid4()}")
+
+
+class BookingError(Exception):
+    ...
+
+
+@lambda_handler_decorator
+def obfuscate_sensitive_data(handler, event, context, fields: List) -> Callable:
+    # extracting payload from a EventBridge event
+    detail: dict = extract_data_from_envelope(data=event, envelope=envelopes.EVENTBRIDGE)
+    guest_data: Any = detail.get("guest")
+
+    # Obfuscate fields (email, vat, passport) before calling Lambda handler
+    if fields:
```

Comment:
maybe this condition is unecessary because the next line would just iterate over nothing

Reply:
Agree 100%. I removed the condition